### PR TITLE
Add "check-versions" command

### PR DESCRIPTION
**Pull request by vagishagupta23**
Add a `--check-versions` command that checks the version of suricata
and logs if the versions are up to date, outdated or EOL.

Make sure these boxes are signed before submitting your Pull Request
-- thank you.

- [x] I have read the contributing guide lines at
  https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [x] I have signed the Open Information Security Foundation
  contribution agreement at
  https://suricata-ids.org/about/contribution-agreement/
- [ ] I have updated the user guide (in doc/userguide/) to reflect the
  changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: https://redmine.openinfosecfoundation.org/issues/2341

### CODE REVIEW

**Comment by vagishagupta23** [comment]
@jasonish I have addressed some changes in this PR. Could you please have a look?
I believe `--check-versions` seems to work fine here but using `--check-versions` with `--suricata-version` is failing many cases. What should be done in such case?

File: suricata/update/data/index.py [hunk 1]
```diff
@@ -1 +1 @@
...
...Positive Technologies Attack Detection Team ruleset'}, 'sslbl/ssl-fp-blacklist': {'url': 'https://sslbl.abuse.ch/blacklist/sslblacklist.rules', 'vendor': 'Abuse.ch', 'description': 'The SSL Blacklist (SSLBL) is a project of abuse.ch with the goal of detecting malicious SSL connections, by identifying and blacklisting SSL certificates used by botnet C&C servers. In addition, SSLBL identifies JA3 fingerprints that helps you to detect & block malware botnet C&C communication on the TCP layer.\n', 'license': 'Non-Commercial', 'summary': 'Abuse.ch SSL Blacklist'}, 'tgreen/hunting': {'vendor': 'tgreen', 'description': 'Heuristic ruleset for hunting. Focus on anomaly detection and showcasing latest engine features, not performance.\n', 'license': 'GPLv3', 'url': 'https://raw.githubusercontent.com/travisbgreen/hunting-rules/master/hunting.rules', 'min-version': '4.1.0', 'summary': 'Threat hunting rules'}, 'etnetera/aggressive': {'url': 'https://security.etnetera.cz/feeds/etn_aggressive.rules', 'vendor': 'Etnetera a.s.', 'min-version': '4.0.0', 'license': 'MIT', 'summary': 'Etnetera aggressive IP blacklist'}}, 'version': 1, 'versions': {'suricata': {'recommended': '4.1.4', '4.1': '4.1.4'}}}
```
Comment (jasonish) [inline suricata/update/data/index.py]: Can you do this as a separate commit?  The reason is, we don't want it as the actual feature commit, we'll regenerate this file with a script from the master index and it will get its own commit.

File: suricata/update/main.py [hunk 2]
```diff
@@ -1129,6 +1150,8 @@ def _main():
     
     update_parser.add_argument("--no-merge", action="store_true", default=False,
                                help="Do not merge the rules into a single file")
+    update_parser.add_argument("--check-versions", action="store_true",
+                               help="Check version of suricata.")
```
Comment (jasonish) [inline suricata/update/main.py]: This should be a sub-command instead.  It should be called from the command line like:
```
suricata-update check-versions
```
similar to update-sources, list-sources, etc.

But I also wonder if it should run as part of the default operation as well.  I can see `check-versions` subcommand useful on its own, to run without doing an update.  But maybe we should run this as part of the update as well.  Make it as a subcommand first, then worry about that though.

**Comment by jasonish** [comment]
We should handle the case where the index does not have the version information, to avoid this:
```
Traceback (most recent call last):
  File "./bin/suricata-update", line 33, in <module>
    sys.exit(main.main())
  File "/home/jason/oisf/code/suricata-update/jasonish/suricata/update/main.py", line 1485, in main
    sys.exit(_main())
  File "/home/jason/oisf/code/suricata-update/jasonish/suricata/update/main.py", line 1271, in _main
    check_version(suricata_version)
  File "/home/jason/oisf/code/suricata-update/jasonish/suricata/update/main.py", line 1027, in check_version
    version = index.get_versions()
  File "/home/jason/oisf/code/suricata-update/jasonish/suricata/update/sources.py", line 130, in get_versions
    return self.index["versions"]
KeyError: 'versions'
```

**Comment by vagishagupta23** [comment]
Thanks for the feedback. I'll address these changes in a new PR.

**Comment by vagishagupta23** [comment]
Replace by a new PR #185
